pkgs/development/tools/misc/lsof/default.nix

```diff
@@ -22,12 +25,14 @@ stdenv.mkDerivation rec {
   };
 
   unpackPhase = "tar xvjf $src; cd lsof_*; tar xvf lsof_*.tar; sourceRoot=$( echo lsof_*/); ";
+  
+  patches = [ ./dfile.patch ];
 
-  preBuild = "sed -i Makefile -e 's/^CFGF=/&	-DHASIPv6=1/;';";
+  preBuild = "sed -i Makefile -e 's/^CFGF=/&	-DHASIPv6=1/;' -e s/-lcurses/-lncurses/ ;";
 
   configurePhase = ''
     # Stop build scripts from searching global include paths
-    export LSOF_INCLUDE=/$(md5sum <(echo $name) | awk '{print $1}')
+    # export LSOF_INCLUDE=/$(md5sum <(echo $name) | awk '{print $1}')
```

Comment:
@Mic92, thanks for pointing it out. Actually, I was not sure about the purpose of that line, but now @adnelson's comment below made it a bit more clear.

By assigning no value, it defaults to '/usr/include', and it makes it successfully compiled on OS X.

Is compiling with a header file outside of nix world a bad practice? If so, I should find out another way.